movement/napari/loader_widgets.py

```diff
@@ -146,76 +147,124 @@ def _on_load_clicked(self):
+            "keypoint" in self.properties
+            and len(self.properties["keypoint"].unique()) > 1
         ):
             text_prop = "keypoint"
-        else:
-            text_prop = "individual"
-        props_and_style.set_text_by(prop=text_prop)
+        points_style.set_text_by(property=text_prop)
 
-        # Set color of markers and text
-        color_prop = "individual"
-        n_individuals = len(self.props["individual"].unique())
-        if n_individuals == 1 and "keypoint" in self.props:
-            color_prop = "keypoint"
-        props_and_style.set_color_by(prop=color_prop)
+        # Set property for markers' and text color
+        points_style.set_color_by(
+            property=self.color_property,
+            properties_df=self.properties,
+        )
 
         # Add data as a points layer
-        points_layer = self.viewer.add_points(
-            self.data[bool_not_nan, 1:],
-            **props_and_style.as_kwargs(),
+        self.points_layer = self.viewer.add_points(
+            self.data[self.bool_not_nan, 1:],
+            properties=self.properties.iloc[self.bool_not_nan, :],
+            **points_style.as_kwargs(),
         )
 
-        # Add metadata to the layer
-        points_layer.metadata = {
-            "min_frame_idx": min(self.data[:, 1]),
+        # Add frame range as metadata to the layer
+        # This would be the frame index after transforming to
+        # napari tracks array
+        self.points_layer.metadata = {
             "max_frame_idx": max(self.data[:, 1]),
         }
 
         logger.info("Added tracked dataset as a napari Points layer.")
 
+    def _add_tracks_layer(self):
+        """Add the tracked data to the viewer as a Tracks layer."""
+        # Factorize the color property (required for tracks layer)
+        codes, _ = pd.factorize(self.properties[self.color_property])
+        color_property_factorized = self.color_property + "_factorized"
+        self.properties[color_property_factorized] = codes
+
+        # Define style for tracks layer
+        tracks_style = TracksStyle(
+            name=f"tracks: {self.file_name}",
+            tail_length=int(max(self.data[:, 1])),
+            # Set the tail length to the number of frames in the data.
+            # If the value is over 300, it sets the maximum
+            # tail_length in the slider to the value passed.
+            # It also affects the head_length slider.
+        )
+
+        # Set color by property
+        tracks_style.set_color_by(property=color_property_factorized)
+
+        # Add data as a tracks layer
+        self.tracks_layer = self.viewer.add_tracks(
+            self.data[self.bool_not_nan, :],
+            properties=self.properties.iloc[self.bool_not_nan, :],
+            **tracks_style.as_kwargs(),
+        )
+
+        # Set display checkboxes
+        self.tracks_layer.display_tail = True
+        self.tracks_layer.display_graph = False
```

Comment:
This actually doesn't seem to do anything - any findings on this are more than welcome